### PR TITLE
feat(@cubejs-client/react): Expose progress in useCubeQuery hook

### PR DESCRIPTION
**Pull request by anton164**
**Check List**
- [ ] Tests has been run in packages where changes made if available
- [x] Linter has been run for changed code
- [ ] Tests for the changes have been added if not covered yet
- [x] Docs have been added / updated if required

**Tests:**
There are no tests for this package at the moment, so I wasn't able to add any for this extension.

**Docs:**
I updated the `index.d.ts` file, but have not updated the other docs/examples. Are they automatically generated somehow or shall I do it manually?

**Description of Changes Made (if issue reference is not provided)**
When loading a Cube Query, it might take some due to waiting in the queue or because pre-aggregations are being built.

By exposing the ProgressResponse from the CubeJS API, the UI can share more information with the user when the query is loading.

This allows us to show the following in our UI (big win for our users):
Screenshot 2020-08-13 at 13 41 48

___

I hope you find this to be a useful addition to Cube.JS :)

### CODE REVIEW

File: packages/cubejs-client-react/src/useCubeQuery.js [hunk 1]
```diff
@@ -11,11 +11,21 @@ export default (query, options = {}) => {
...
   const [isLoading, setLoading] = useState(false);
   const [resultSet, setResultSet] = useState(null);
+  const [progress, setProgress] = useState(null);
   const [error, setError] = useState(null);
   const context = useContext(CubeContext);
 
   let subscribeRequest = null;
 
+  const progressCallback = ({ progressResponse }) => {
+    setProgress({
```
Comment (vasilev-alex) [inline packages/cubejs-client-react/src/useCubeQuery.js:21]: Let's just save the `ProgressResponse` directly and let the user customize it on their side

**Review by vasilev-alex — CHANGES_REQUESTED** [review]
Hey @anton164! Thanks for the PR

File: packages/cubejs-client-react/src/useCubeQuery.js (hunk 1 above)
Comment (anton164) [reply packages/cubejs-client-react/src/useCubeQuery.js:21]: I can do that, but then the code for consuming the progress response will be much more convoluted?

I.e. the caller needs to access `.stage.stage` to get the message 🤷‍♂️ 

```
 export default function App() {
   *   const { resultSet, isLoading, error } = useCubeQuery({
   *   const { resultSet, isLoading, error, progress } = useCubeQuery({
   *     measures: ['Orders.count'],
   *     dimensions: ['Orders.createdAt.month'],
   *   });
   *
   *   if (isLoading) {
   *     return <div>Loading...</div>;
   *     return <div>{progress && progress.stage && progress.stage.stage || 'Loading...'}</div>;
   *   }
```

Comment (anton164) [reply packages/cubejs-client-react/src/useCubeQuery.js]: Updated now 👍

**Pull request merged**